packages/react-scripts/template/src/registerServiceWorker.js

```diff
@@ -56,12 +95,12 @@ function registerValidSW(swUrl) {
               // the fresh content will have been added to the cache.
               // It's the perfect time to display a "New content is
               // available; please refresh." message in your web app.
-              console.log('New content is available; please refresh.');
+              showMessage('New content is available; please refresh.');
```

Comment:
Possible UX improvements:

1. add link which user can click to reload page:

```html
New content is available, <a onclick="window.location.reload()" >please refresh</a>.
```

2. Do not show message about reload and instead reload browser when user navigates to the next page. But this makes sense only if app actually has routes and uses something like react-router

Reply:
nice idea! I am waiting for https://github.com/facebookincubator/create-react-app/pull/2755 to get in so we can solve https://github.com/facebookincubator/create-react-app/issues/2638 and continue working on SW enhancements

Reply:
#2755 closed BTW